sstables/sstables.cc

```diff
@@ -3140,6 +3140,21 @@ std::optional<large_data_stats_entry> sstable::get_large_data_stat(large_data_ty
     return std::make_optional<large_data_stats_entry>();
 }
 
+malformed_sstable_exception::malformed_sstable_exception(const sstable& sst, sstring msg, const sstring& filename)
+    : malformed_sstable_exception(fmt::format("{} in sstable {} origin={}", msg, filename, sst.get_origin()))
+{
+}
+
+[[noreturn]] void throw_malformed_sstable_exception(logging::logger& log, const sstable& sst, sstring msg, const sstring& filename) {
```

Comment:
With a fresh mind in a fresh year I realized that nothing stops new code from just using `throw malformed_sstable_exception` as before, completely circumventing this new feature. If covering all throw sites both present and future is your intention, this might be better served by making a logger a mandatory constructor argument of `malformed_sstable_exception` and doing the logging in the constructor. That way all future code will automatically benefit from this patch.

Reply:
That would work, though I think that passing a logger to the exception constructor is quite ugly.
I can also rename the exception type to e.g. `_malformed_sstable_exception` so it would be clearer that it's an internal name.

Plus, there was at least one call site, in `sstable::read_simple` where I left the code unchanged, as follows:
```
        } catch (malformed_sstable_exception &e) {
            throw malformed_sstable_exception(e.what(), file_path);
        }
```

since the code was just wrapping a nested malformed_sstable_exception and adding the file_path, since lower-level functions that are static are not aware of the sstable file_path.